src/main/java/bio/terra/service/dataset/flight/datadelete/CreateExternalTablesStep.java

```diff
@@ -0,0 +1,115 @@
+package bio.terra.service.dataset.flight.datadelete;
+
+import bio.terra.common.exception.PdaoInvalidUriException;
+import bio.terra.common.exception.PdaoSourceFileNotFoundException;
+import bio.terra.model.DataDeletionRequest;
+import bio.terra.model.DataDeletionTableModel;
+import bio.terra.service.dataset.Dataset;
+import bio.terra.service.dataset.DatasetService;
+import bio.terra.service.dataset.exception.InvalidFileRefException;
+import bio.terra.service.dataset.exception.TableNotFoundException;
+import bio.terra.service.dataset.flight.ingest.IngestUtils;
+import bio.terra.service.filedata.google.gcs.GcsPdao;
+import bio.terra.service.tabulardata.google.BigQueryPdao;
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import com.google.cloud.storage.Storage;
+import com.google.cloud.storage.StorageOptions;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getDataset;
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getRequest;
+import static bio.terra.service.dataset.flight.datadelete.DataDeletionUtils.getSuffix;
+
+
+public class CreateExternalTablesStep implements Step {
+
+    private final BigQueryPdao bigQueryPdao;
+    private final DatasetService datasetService;
+    private final Storage storage;
+
+    private static Logger logger = LoggerFactory.getLogger(CreateExternalTablesStep.class);
+
+    public CreateExternalTablesStep(BigQueryPdao bigQueryPdao, DatasetService datasetService) {
+        this.bigQueryPdao = bigQueryPdao;
+        this.datasetService = datasetService;
+        this.storage = StorageOptions.getDefaultInstance().getService();
+    }
+
+    private boolean fileExistsForTable(DataDeletionTableModel table) {
+        // TODO: support other file types (assuming header-less csv here)
+        String path  = table.getGcsFileSpec().getPath();
+        IngestUtils.GsUrlParts gsUrlParts = IngestUtils.parseBlobUri(path);
+        try {
+            if (gsUrlParts.getIsWildcard()) {
+                // There's not an easy way to check this with the SDK. Let BQ handle catching this on its side.
+                return true;
+            } else {
+                return GcsPdao.getBlobFromGsPath(storage, path).exists();
```

Comment:
If we're ok letting BQ be the failure point for the wildcard case, are we also ok letting it be the same for the single-file case? It'd save us up-front work / API calls.